WordPress/dashicons
# WebFont: remove external dependencies (no fontforge, no ttfautohint)

This PR tries to remove external dependencies (`brew` + `fontforge` + `ttfautohint`), using exclusively the `node` rendering engine. 

Settings to make this work:

* engine: node (no`fontforge` dependency, [docs](https://github.com/sapegin/grunt-webfont#available-engines))
* autoHint: false (no `ttfautohint` dependency)
* normalize: true (fixes sizing issue)
* optimize: false (fixed deformed shapes)

Even without optimization, the file size seems smaller (TTF:  59k → 42k).

It seems I managed the icons to render, but I need expert eyes to cross check accurately everything. A cursory view of `dashicons.html` seems alright: 

![screen shot 2017-07-07 at 18 10 49](https://user-images.githubusercontent.com/4389/27968625-afcc6740-633f-11e7-9884-9c909681c99e.png)

In the process I dug a fair bit in the grunt, npm, fontforge ecosystem and appears we're on a bit of a shaky ground as the one we use is the only project around that is somehow maintained, and even there the switch to SVG font seems having made the interest in this library diminish over time too. After this work I feel we should try to switch to SVG as soon as possible, but this is a different topic.


### How to test?

1. The build process should be as usual. 
2. Compare the icons from this TTF to the icons from the original, both in `dashicons.html` and in a WordPress install.
